java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -869,28 +869,75 @@ public static HostColumnVector timestampNanoSecondsFromBoxedLongs(Long... values
     private boolean nullable;
     private long rows;
     private long estimatedRows;
+    private long rowCapacity = 0L;
+    private long validCapacity = 0L;
     private boolean built = false;
     private List<ColumnBuilder> childBuilders = new ArrayList<>();
+    private Runnable nullHandler;
 
     private int currentIndex = 0;
-    private int currentByteIndex = 0;
-
+    private long currentByteIndex = 0;
 
     public ColumnBuilder(HostColumnVector.DataType type, long estimatedRows) {
       this.type = type.getType();
       this.nullable = type.isNullable();
       this.rows = 0;
       this.estimatedRows = estimatedRows;
+
+      // initialize the null handler according to the data type
+      this.setupNullHandler();
+
       for (int i = 0; i < type.getNumChildren(); i++) {
         childBuilders.add(new ColumnBuilder(type.getChild(i), estimatedRows));
       }
     }
 
+    private void setupNullHandler() {
+      if (this.type == DType.LIST) {
+        this.nullHandler = () -> {
+          this.growListBuffersAndRows();
+          this.growValidBuffer();
+          setNullAt(currentIndex++);
+          currentByteIndex += this.type.getSizeInBytes();
```

Comment:
Before we set this for all of the null handling code, but I don't see any reason to do it. I would just not set it at all for anything but a String, and even then we don't need to add anything to it because the sizeInBytes for a String column returns 0.

Reply:
Done.